pkg/system_api.go

```diff
@@ -91,36 +147,37 @@ func (components *Components) UnmarshalJSON(b []byte) error {
 }
 
 type Component struct {
-	Name         string            `json:"name,omitempty"`
-	Id           string            `json:"id,omitempty"` //nolint
-	Text         string            `json:"text,omitempty"`
-	TopologyType string            `json:"topologyType,omitempty"`
-	Namespace    string            `json:"namespace,omitempty"`
-	Labels       map[string]string `json:"labels,omitempty"`
-	Tooltip      string            `json:"tooltip,omitempty"`
-	Icon         string            `json:"icon,omitempty"`
-	Owner        string            `json:"owner,omitempty"`
-	Status       string            `json:"status,omitempty"`
-	StatusReason string            `json:"statusReason,omitempty"`
+	Name         string              `json:"name,omitempty"`
+	ID           uuid.UUID           `json:"id,omitempty" gorm:"default:generate_ulid()"` //nolint
+	Text         string              `json:"text,omitempty"`
+	TopologyType string              `json:"topologyType,omitempty"`
+	Namespace    string              `json:"namespace,omitempty"`
+	Labels       types.JSONStringMap `json:"labels,omitempty"`
+	Tooltip      string              `json:"tooltip,omitempty"`
+	Icon         string              `json:"icon,omitempty"`
+	Owner        string              `json:"owner,omitempty"`
+	Status       string              `json:"status,omitempty"`
+	StatusReason string              `json:"statusReason,omitempty"`
 	// The type of component, e.g. service, API, website, library, database, etc.
 	Type    string     `json:"type,omitempty"`
-	Summary v1.Summary `json:"summary,omitempty"`
+	Summary v1.Summary `json:"summary,omitempty" gorm:"type:summary"`
 	// The lifecycle state of the component e.g. production, staging, dev, etc.
-	Lifecycle     string             `json:"lifecycle,omitempty"`
-	Relationships []RelationshipSpec `json:"relationships,omitempty"`
-	Properties    Properties         `json:"properties,omitempty"`
-	Components    Components         `json:"components,omitempty"`
-	ParentId      string             `json:"parent_id,omitempty"` //nolint
-	CreatedAt     string             `json:"created_at,omitempty"`
-	UpdatedAt     string             `json:"updated_at,omitempty"`
-	ExternalId    string             `json:"external_id,omitempty"` //nolint
+	Lifecycle     string        `json:"lifecycle,omitempty"`
+	Relationships Relationships `json:"relationships,omitempty" gorm:"type:relationships"`
```

Comment:
We can't embed this as a json into component - It will cause very high locking / deadlocking on the row when multiple components needed to be added, it will also make querying difficult as the component Id's will need to be extracted from the JSON . 

We would need s seperate table e.g. 

parentId 
compononentId
type
createdAt
systemTemplateId
updatedAt (Every topology run should refresh this, so that stale relationships can be removed)

- We should however move this to a subsequent PR.